narwhal/types/src/metered_channel.rs

```diff
@@ -32,6 +36,7 @@ impl<T> Clone for Sender<T> {
 pub struct Receiver<T> {
     inner: mpsc::Receiver<T>,
     gauge: IntGauge,
+    total: Option<IntGauge>,
```

Comment:
I believe for the nature of the metrics we are looking at (a monotonically increasing counter) it would be better to use the [IntCounter](https://docs.rs/prometheus/latest/prometheus/type.IntCounter.html) type instead.